pkg/broker/config/memory/memory.go

```diff
@@ -81,22 +80,14 @@ var _ config.Targets = (*memoryTargets)(nil)
 
 // NewEmptyTargets returns an empty mutable Targets in memory.
 func NewEmptyTargets() config.Targets {
-	t := &memoryTargets{
-		mux: sync.Mutex{},
-	}
-	t.Store(&config.TargetsConfig{Brokers: make(map[string]*config.Broker)})
-	return t
+	return NewTargets(&config.TargetsConfig{Brokers: make(map[string]*config.Broker)})
 }
 
-// NewTargetsFromBytes creates a mutable Targets in memory.
-func NewTargetsFromBytes(b []byte) (config.Targets, error) {
```

Comment:
This could be useful when reading bytes from configmap to form a memory config. I don't see it's used by the controller though. I'm fine replacing (vs. adding a new func) if it's not used. @grantr

Reply:
The broker should be reading the configuration from the configmap on start. It doesn't do that yet but it will need to soon. Otherwise the configmap will be inaccurate while the initial resync happens.

Reply:
In other words, the controller will need some way to load an in-memory targets config from a configmap (not a volume).

Reply:
I guess option 1: keep `FromBytes` and the controller can pass the bytes from configmap; or option 2: replace this `FromBytes` with `FromConfigMap` which accepts a configmap object.

Reply:
I don't like option 2. It requires this code to be aware that the config comes from a configmap and what filename the data is in. That seems like the wrong layering.

I'm fine with option 1 or this PR. I think option 1 is slightly easier since it removes any proto marshaling from the controller code and tests.